adafruit_azureiot/device_registration.py

```diff
@@ -45,8 +44,8 @@ class DeviceRegistration:
     # pylint: disable=R0913
     def __init__(
         self,
-        socket,
-        iface,
+        socket_pool,
```

Comment:
Change `__init__` to take in `socket_pool` (renamed) and `ssl_context`. Use it to pass to `MQTT`